src/test/java/ca/uhn/fhir/jpa/starter/OperationCollectDataTest.java

```diff
@@ -142,10 +270,28 @@ private Parameters fetchParameter(String theUrl) throws IOException {
 		return parameters;
 	}
 
+	private Parameters fetchXmlParameter(String theUrl) throws IOException {
```

Comment:
So, I'd suggest combining `fetchParameter()` and `fetchXmlParameter()` into something like this:

```java
public static final MediaType FHIR_JSON_MEDIA_TYPE = MediaType.valueOf("application/fhir+json");
public static final MediaType FHIR_XML_MEDIA_TYPE = MediaType.valueOf("application/fhir+xml");

// ...

private Parameters fetchParameters(String theUrl, MediaTypes mediaType) throws IOException {
	HttpGet get = new HttpGet(theUrl);

	String auth = USER_NAME + ":" + USER_PASSWORD;
	byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
	String authHeader = "Basic " + new String(encodedAuth);
	get.addHeader(HttpHeaders.AUTHORIZATION, authHeader);
	get.addHeader(HttpHeaders.ACCEPT, mediaType.toString());


	try (CloseableHttpResponse resp = ourHttpClient.execute(get)) {
		IParser parser;
		MediaType responseMediaType = MediaType.parseMediaType(response.getContentType());
        if (FHIR_JSON_MEDIA_TYPE.isCompatibleWith(responseMediaType) || MediaType.APPLICATION_JSON.isCompatibleWith(responseMediaType)) {
			parser = ourCtx.newJsonParser();
		} else if (FHIRXML_MEDIA_TYPE.isCompatibleWith(responseMediaType) || MediaType.APPLICATION_XML.isCompatibleWith(responseMediaType) || MediaType.TEXT_XML.isCompatibleWith(responseMediaType)) {
			parser = ourCtx.newXmlParser();
		} else {
			throw AssertionError("Cannot handle response of type " + responseMediaType.toString());
		}

		return parser.parseResource(Parameters.class,
			  EntityUtils.toString(resp.getEntity(), Charsets.UTF_8));
	}
	
	throw new AssertionError("Could not retrieve parameters from server");
}
```

Then use just changes to:

```java
fetchParameters(theUrl, FHIR_JSON_MEDIA_TYPE);
// or
fetchParameters(theUrl, FHIR_XML_MEDIA_TYPE);
```

I'd do something similar with `postResource()` and `postXmlResource()`